### PR TITLE
Define operators == and != for ItemStack

### PR DESCRIPTION
**Pull request by ClobberXD**
- `==` operator returns `true` if the name, count, wear, _and_ metadata of the two stacks are the same.
- `!=` operator uses `==` internally, and returns the inverse of the result. i.e. `return !(s1 == s2);`

I used the following regex pattern to check for itemstack comparisons, and couldn't find any:
```regex
\w+(->|\.)name != \w+(->|\.)name
```

Therefore, this PR is complete.

Use-cases:

- ItemStack comparison for whatever reason :)
- A new API method `ItemStack:equals` could be implemented later on.

See IRC logs for related discussion - http://irc.minetest.net/minetest-dev/2019-05-05#i_5540154

### CODE REVIEW

File: src/inventory.h [hunk 1]
```diff
@@ -161,6 +161,22 @@ struct ItemStack
 	// Similar to takeItem, but keeps this ItemStack intact.
 	ItemStack peekItem(u32 peekcount) const;
 
+	bool operator ==(const ItemStack &s) const
+	{
+		if (this->name != s.name ||
```
Comment (stujones11) [inline src/inventory.h]: why not just return the condition directly? Also `==` and `&&` would seem more logical here

**Comment by paramat** [comment]
To be clear about the need: I seem to remember you discovered that these operators are defined many times throughout MT code, so this would lead to removing a lot of duplicated code? Am i right?

**Comment by rubenwardy** [comment]
> To be clear about the need: I seem to remember you discovered that these operators are defined many times throughout MT code, so this would lead to removing a lot of duplicated code? Am i right?

No, this is not correct. He believed this because of an error message he got when trying to implement the operator.

This operator is needed because it makes the code much nicer, and is how you should do comparison in C++. Note that the compiler will inline this anyway, so there's no disadvantages

File: src/inventory.h (hunk 1 above)
Comment (ClobberXD) [reply src/inventory.h]: Good idea :+1:

**Comment by ClobberXD** [comment]
@paramat That was because I defined the `!=` operator in the header, as a standalone function, which resulted in the multiple function defs warning. I was surprised to see that this wasn't used anywhere at all (except in inventory.cpp at around L550).

File: src/inventory.h (hunk 1 above)
Comment (ClobberXD) [reply src/inventory.h]: Done. Code's now much simpler, thanks. :)

**Comment by paramat** [comment]
Ok thanks.
There is no issue for this PR (which is fine), so the first post should have contained an explanation of why this is needed / useful. There was an IRC discussion (a link to logs would be nice) but many core devs will not have seen that.

**Comment by ClobberXD** [comment]
Updated first post ;)

**Comment by ClobberXD** [comment]
Hmm, I assumed braces were compulsory for loops, thanks.

**Pull request merged**